File: leetcode/lenlastword/lenlastword.c
int lengthOfLastWord(char * s){
    int prev_count = 0, count = 0;
    while (*s != '\0') {
        if (*s == ' ') {
            if (count)
                prev_count = count;
            count = 0;
        } else {
            count++;
        }
        s++;
    }
    return (count == 0 ? prev_count : count);
}

File: leetcode/lenlastword/lenlastword.py
class Solution(object):
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        count = 0
        prev_count = 0
        if len(s) < 1:
            return count
        for let in s:
            if let == ' ':
                if count <> 0:
                    prev_count = count
                count = 0
            else:
                count += 1

        if count == 0:
            return prev_count
        return count
